grpc/src/main/java/com/linecorp/armeria/server/grpc/GrpcServiceBuilder.java

```diff
@@ -107,11 +107,13 @@ public GrpcServiceBuilder addService(ServerServiceDefinition service) {
      */
     public GrpcServiceBuilder addService(BindableService bindableService) {
         if (bindableService instanceof ProtoReflectionService) {
-            checkState(protoReflectionService == null,
+            checkState(!isProtoReflectionServiceSet,
                        "Attempting to add a ProtoReflectionService but one is already present. " +
                        "ProtoReflectionService must only be added once.");
-            protoReflectionService = (ProtoReflectionService) bindableService;
-        }
+            isProtoReflectionServiceSet = true;
+            return addService(ServerInterceptors.intercept(bindableService,
```

Comment:
Instead of adding here, can't we add in the same place we called `notifyOnBuild` before so we can use the static server config instead of request context?

Reply:
Ah nevermind guess we can't add more services there :)

Reply:
The creatin logic is moved back to FramedGrpcService. 😀